gears/src/types/context/mod.rs

```diff
@@ -1,5 +1,54 @@
-pub mod context;
+use database::{Database, PrefixDB};
+use proto_messages::cosmos::tx::v1beta1::tx_metadata::Metadata;
+use store_crate::{ReadKVStore, WriteKVStore};
+use tendermint::informal::{abci::Event, chain::Id};
+
 pub mod init_context;
 pub mod query_context;
-pub mod read_context;
 pub mod tx_context;
+
+pub trait Context<DB, SK> {
```

Comment:
I'm think that `Context` name is wrong and it should be called QueryableContext. What do you think?

Reply:
Matter of taste. I use separate trait 'cause I don't want trait bounds to `Database & StoreKey` + as I remember, there should be cases where we only need access to KVStore.

Reply:
But I don't think we'll ever have a type which implements the `ReadTrait` and doesn't implement the `Context` trait.

Reply:
Unified this trait with `ReadContext`